examples/rollout-template-ref.yaml

```diff
@@ -0,0 +1,42 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app.kubernetes.io/instance: rollout-canary
+  name: rollout-ref-deployment
+spec:
+  replicas: 0
+  selector:
+    matchLabels:
+      app: rollout-ref-deployment
+  template:
+    metadata:
+      labels:
+        app: rollout-ref-deployment
+    spec:
+      containers:
+        - name: rollouts-demo
+          image: argoproj/rollouts-demo:blue
+          imagePullPolicy: Always
+          ports:
+            - containerPort: 8080
+---
+apiVersion: argoproj.io/v1alpha1
+kind: Rollout
+metadata:
+  name: rollout-ref-deployment
+spec:
+  replicas: 5
+  revisionHistoryLimit: 2
+  selector:
+    matchLabels:
+      app: rollout-ref-deployment
+  templateRef:
```

Comment:
I think `workloadRef` may be better than `templateRef`

Reply:
I've changed `workloadRef` to `templateRef` to make it more explicit. Howerver, after looking at comment https://github.com/argoproj/argo-rollouts/pull/1072#discussion_r610208462, I agree. Changing back to `workloadRef`

Reply:
done